src/pem.c

```diff
@@ -116,7 +126,45 @@ _libssh2_pem_parse(LIBSSH2_SESSION * session,
     }
     while (strcmp(line, headerbegin) != 0);
 
-    *line = '\0';
+    if (readline(line, LINE_SIZE, fp)) {
+        return -1;
+    }
+
+    const LIBSSH2_CRYPT_METHOD *method = NULL;
+    if (passphrase &&
+            memcmp(line, crypt_annotation, strlen(crypt_annotation)) == 0) {
+        if (readline(line, LINE_SIZE, fp)) {
+            ret = -1;
+            goto out;
+        }
+        const LIBSSH2_CRYPT_METHOD **all_methods = libssh2_crypt_methods();
+        const LIBSSH2_CRYPT_METHOD *cur_method;
```

Comment:
These variable definitions should be at the start of the scope to avoid problems with strict C89/C90 compilers.